docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,220 @@
+
+Given the code below, we use the `map()` function to take an array of `numbers`, double their values, and then assigned the new array to the variable `doubled`.
+
+```javascript
+const numbers = [1, 2, 3, 4, 5];
+const doubled = numbers.map((number) => number * 2); // array: [2, 4, 6, 8, 10]
+```
+
+In React, transforming arrays into lists of [elements](/react/docs/rendering-elements.html) is nearly identical.
+
+### Rendering Multiple Components
+
+You can build collections of elements and include them in [JSX](/react/docs/introducing-jsx.html) using curly braces `{}`, similar to embedding values with Javascript.
+
+Below, we loop through the `numbers` array using the Javascript [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) function. We return an `<li>` element for each item. Finally, we assign the resulting array of items to `listItems`. In `ReactDOM.render()` we render the entire `{listItems}` array inside a `<ul>` element.
+
+```javascript
+const numbers = [1, 2, 3, 4, 5];
+const listItems = numbers.map((item) => <li>{item}</li>);
+ReactDOM.render(<ul>{listItems}</ul>, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/wzxEmv/?editors=0011)
+
+### Basic List Component
+
+We can refactor the previous example into a [functional component](http://localhost:4000/react/docs/components-and-props.html) that accepts an array of `numbers` and outputs an unordered list of elements.
+
+```javascript
+function NumberList(props) {
+  const numbers = props.numbers
+  const listItems = numbers.map((item) => <li>{item}</li>);
+  return <ul>{listItems}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+When you run this code, you'll be given a warning that a key should be provided for list items. Keys should be included when creating lists of elements.
+
+Let's assign a `key` to our list items inside `numbers.map()` and fix the missing key issue.
+
+```javascript{11}
+function NumberList(props) {
+  const numbers = props.numbers
+  const listItems = numbers.map((item) => <li key={"item-" + item}>{item}</li>);
+  return <ul>{listItems}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/gwjdzN?editors=0011)
+
+
+### Keys
+
+Keys are important for helping React know if items have changed, are added, or removed.
+
+Use keys that represent the identity of the item. If you don't have a way to uniquely identify each item consider using a string representation of the item. _You will be warned if you don't include keys._
+
+Supply keys to elements inside a `map()` function and not directly to an HTML element. Keys used within groups should be unique to each other. Keys cannot be accessed as a `prop`.
```

Comment:
The problem is not that it's "directly to an HTML element". (This could actually be a component element too and the rule would still apply.)

Maybe say "Keys should be given to the elements inside the array because they give the elements a stable identity. If you extract a `Number` component, you should keep the `key` on the `<Number />` elements in the array rather than on the root `<li>` element in the `Number` itself."